### PR TITLE
Support multiple network ids and network names

### PR DESCRIPTION
**Pull request by sue445**
CloudStack `network_ids` can accept multiple values. But vagrant-cloudstack plugin can not use multiple network_ids (only single network_id).

So I added feature for multiple network ids
# Usage

Vagrantfile

``` ruby
Vagrant.configure(2) do |config|
  config.vm.provider :cloudstack do |cloudstack, override|
    # single network id
    cloudstack.network_id = "AAAAA"

    # multiple network ids
    cloudstack.network_id = ["AAA", "BBB"]
  end
end
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -126,7 +126,8 @@ to update UUIDs in your Vagrantfile. If both are specified, the id parameter tak
 * `instance_ready_timeout` - The number of seconds to wait for the instance
   to become "ready" in Cloudstack. Defaults to 120 seconds.
 * `domain_id` - Domain id to launch the instance into
-* `network_id` - Network uuid that the instance should use
+* `network_id` - Network uuid(s) that the instance should use
+  - `network_id` is single value (e.g. `"AAAA"`) or multiple values (e.g. `["AAAA", "BBBB"]`)
 * `network_name` - Network name that the instance should use
```
Comment (bheuvel) [inline README.md]: @sue445 Nice one! But if we do multiple networks by `network_id`, we should also allow that via `network_name`

File: lib/vagrant-cloudstack/action/run_instance.rb [hunk 2]
```diff
@@ -34,8 +34,15 @@ def call(env)
...
+          @network_ids =
+            if @domain_config.network_id.nil?
+              []
+            else
+              Array(@domain_config.network_id)
+            end
+
           @zone             = CloudstackResource.new(@domain_config.zone_id, @domain_config.zone_name, 'zone')
-          @network          = CloudstackResource.new(@domain_config.network_id, @domain_config.network_name, 'network')
+          @network          = CloudstackResource.new(@network_ids.first, @domain_config.network_name, 'network')
```
Comment (bheuvel) [inline lib/vagrant-cloudstack/action/run_instance.rb]: The `@network` object is used to translate `id` from/to `name`. Probably we need an array of `@network` objects which is filled by the (array or single instance) of names and/or IDs.

File: lib/vagrant-cloudstack/action/run_instance.rb [hunk 3]
```diff
@@ -205,7 +212,7 @@ def create_vm
                 :image_id => @template.id
             }
 
-            options['network_ids'] = @network.id unless @network.id.nil?
+            options['network_ids'] = @network_ids.join(",") unless @network_ids.empty?
```
Comment (bheuvel) [inline lib/vagrant-cloudstack/action/run_instance.rb]: By not using `@network` you are effectively disabling specifying usage of `network_name`

File: lib/vagrant-cloudstack/action/run_instance.rb (hunk 2 above)
Comment (sue445) [inline lib/vagrant-cloudstack/action/run_instance.rb]: I see. I'll fix, thank you!

**Comment by sue445** [comment]
@bheuvel I pushed fixes. Please review again :bow:

File: lib/vagrant-cloudstack/action/run_instance.rb [hunk 4]
```diff
@@ -34,8 +34,22 @@ def call(env)
 
           sanitize_domain_config
 
+          network_ids =
```
Comment (bheuvel) [inline lib/vagrant-cloudstack/action/run_instance.rb]: Perhaps you can move this new code block (lines 37-49) to the `sanitize_domain_config`?

**Comment by bheuvel** [comment]
It's looking good! But I will need to have a closer look later on, so some comments after a quick look:

We may need to support either IDs OR Names and not both at the same time.
This may become a problem is someone would specify IDs `id1 id2` and names `name3 name2 name1`. Due to the sync having a "preference" for IDs, the resulting sync would have `id1 id2 id1`.

I would suggest to keep it simple and use IDs if present, and only use names if IDs are not present.
So basically `@domain_config.network_name = [] unless @domain_config.network_id.nil?`
(which could also go into the `sanitize_domain_config`)
That would also simplify the `create_list` (and drop the `self.`?)

**Comment by sue445** [comment]
> I would suggest to keep it simple and use IDs if present, and only use names if IDs are not present.

@bheuvel I fix and push.

**Comment by bheuvel** [comment]
@sue445 All tests passing, did some "manual" testing with multiple network names/ids, and is looking good! Thanks!

**Comment by sue445** [comment]
:smiley:
